app/controllers/declaration/SealController.scala

```diff
@@ -108,12 +108,16 @@ class SealController @Inject()(
     formWithErrors: Form[Seal],
     cachedSeals: Seq[Seal]
   )(implicit request: JourneyRequest[_], appConfig: AppConfig) =
-    cacheService
+    customsCacheService
       .fetchAndGetEntry[TransportDetails](cacheId, TransportDetails.formId)
       .map(data => BadRequest(sealPage(formWithErrors, cachedSeals, data.fold(false)(_.container))))
 
   private def cacheAndRedirect(seals: Seq[Seal])(implicit request: JourneyRequest[_]): Future[Result] =
-    cacheService
-      .cache[Seq[Seal]](cacheId, formId, seals)
-      .map(_ => Redirect(routes.SealController.displayForm()))
+    for {
+      _ <- updateCache(journeySessionId, seals)
+      _ <- customsCacheService.cache[Seq[Seal]](cacheId, formId, seals)
+    } yield Redirect(routes.SealController.displayForm())
```

Comment:
similar to the above, you can move the for comprehension into `updateCache` (removes the duplication and makes  it easier to remove the old cache

Reply:
This has been addressed!